provider/docker/config_test.go

```diff
@@ -0,0 +1,2138 @@
+			if test.expected.error {
+				require.Error(t, actualError)
+			} else {
+				require.NoError(t, actualError)
+			}
+			assert.Equal(t, test.expected.ip, actualIP)
+			assert.Equal(t, test.expected.port, actualPort)
+		})
+	}
+}
+
+func TestDockerGetPort(t *testing.T) {
+	testCases := []struct {
+		desc       string
+		container  docker.ContainerJSON
+		serverPort string
+		expected   string
+	}{
+		{
+			desc:      "no binding, no server port label",
+			container: containerJSON(name("foo")),
+			expected:  "",
+		},
+		{
+			desc: "binding, no server port label",
+			container: containerJSON(ports(nat.PortMap{
+				"80/tcp": {},
+			})),
+			expected: "80",
+		},
+		{
+			desc: "binding, multiple ports, no server port label",
+			container: containerJSON(ports(nat.PortMap{
+				"80/tcp":  {},
+				"443/tcp": {},
+			})),
+			expected: "80",
+		},
+		{
+			desc:       "no binding, server port label",
+			container:  containerJSON(),
+			serverPort: "8080",
+			expected:   "8080",
+		},
+		{
+			desc: "binding, server port label",
+			container: containerJSON(
+				ports(nat.PortMap{
+					"80/tcp": {},
+				})),
+			serverPort: "8080",
+			expected:   "8080",
+		},
+		{
+			desc: "binding, multiple ports, server port label",
+			container: containerJSON(ports(nat.PortMap{
+				"8080/tcp": {},
+				"80/tcp":   {},
+			})),
+			serverPort: "8080",
+			expected:   "8080",
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
+
+			dData := parseContainer(test.container)
+
+			actual := getPort(dData, test.serverPort)
+			assert.Equal(t, test.expected, actual)
+		})
+	}
+}
+
+func TestDockerGetIPAddress(t *testing.T) {
+	testCases := []struct {
+		desc      string
+		container docker.ContainerJSON
+		network   string
+		expected  string
+	}{
+		{
+			desc:      "one network, no network label",
+			container: containerJSON(withNetwork("testnet", ipv4("10.11.12.13"))),
+			expected:  "10.11.12.13",
+		},
+		{
+			desc: "one network, network label",
+			container: containerJSON(
+				withNetwork("testnet", ipv4("10.11.12.13")),
+			),
+			network:  "testnet",
+			expected: "10.11.12.13",
+		},
+		{
+			desc: "two networks, network label",
+			container: containerJSON(
+				withNetwork("testnet", ipv4("10.11.12.13")),
+				withNetwork("testnet2", ipv4("10.11.12.14")),
+			),
+			network:  "testnet2",
+			expected: "10.11.12.14",
+		},
+		{
+			desc: "two networks, no network label, mode host",
+			container: containerJSON(
+				networkMode("host"),
+				withNetwork("testnet", ipv4("10.11.12.13")),
+				withNetwork("testnet2", ipv4("10.11.12.14")),
+			),
+			expected: "127.0.0.1",
+		},
+		{
+			desc: "two networks, no network label, mode host, use provider network",
+			container: containerJSON(
+				networkMode("host"),
+				withNetwork("testnet", ipv4("10.11.12.13")),
+				withNetwork("webnet", ipv4("10.11.12.14")),
+			),
+			expected: "10.11.12.14",
+		},
+		{
```

Comment:
Is it a duplicate of the test L1969?

Reply:
nope

Reply:
I don't know if it is important but don't we have to write a different test description ?